skimage/segmentation/_chan_vese.py

```diff
@@ -0,0 +1,323 @@
+import numpy as np
+from scipy.ndimage import distance_transform_edt as distance
+
+
+def _cv_curvature(phi):
+    """
+    Returns the 'curvature' of a level set 'phi'.
+    """
+    P = np.pad(phi, 1, mode='edge')
+    fy = (P[2:, 1:-1] - P[0:-2, 1:-1]) / 2.0
+    fx = (P[1:-1, 2:] - P[1:-1, 0:-2]) / 2.0
+    fyy = P[2:, 1:-1] + P[0:-2, 1:-1] - 2*phi
+    fxx = P[1:-1, 2:] + P[1:-1, 0:-2] - 2*phi
+    fxy = .25 * (P[2:, 2:]+P[:-2, :-2]-P[:-2, 2:]-P[2:, :-2])
+    grad2 = fx**2 + fy**2
+    K = ((fxx*fy**2 - 2*fxy*fx*fy + fyy*fx**2) /
+         (grad2*np.sqrt(grad2) + 1e-10))
+    return K
+
+
+def _cv_calculate_variation(img, phi, mu, lambda1, lambda2, dt):
+    """
+    Returns the variation of level set 'phi' based on algorithm
+    parameters.
+    """
+    eta = 1e-16
+    P = np.pad(phi, 1, mode='edge')
+
+    phixp = P[1:-1, 2:] - P[1:-1, 1:-1]
+    phixn = P[1:-1, 1:-1] - P[1:-1, :-2]
+    phix0 = (P[1:-1, 2:] - P[1:-1, :-2]) / 2.0
+
+    phiyp = P[2:, 1:-1] - P[1:-1, 1:-1]
+    phiyn = P[1:-1, 1:-1] - P[:-2, 1:-1]
+    phiy0 = (P[2:, 1:-1] - P[:-2, 1:-1]) / 2.0
+
+    C1 = 1. / np.sqrt(eta + phixp**2 + phiy0**2)
+    C2 = 1. / np.sqrt(eta + phixn**2 + phiy0**2)
+    C3 = 1. / np.sqrt(eta + phix0**2 + phiyp**2)
+    C4 = 1. / np.sqrt(eta + phix0**2 + phiyn**2)
+
+    K = (P[1:-1, 2:]*C1 + P[1:-1, :-2]*C2 +
+         P[2:, 1:-1]*C3 + P[:-2, 1:-1]*C4)
+
+    Hphi = 1 * (phi>0)
+    (c1, c2) = _cv_calculate_averages(img, Hphi)
+
+    difference_from_average_term = (- lambda1*(img-c1)**2 +
+                                    lambda2*(img-c2)**2)
+    new_phi = (phi + (dt*_cv_delta(phi)) *
+               (mu*K + difference_from_average_term))
+    return new_phi / (1 + mu*dt*_cv_delta(phi)*(C1+C2+C3+C4))
+
+
+def _cv_heavyside(x, eps=1.):
+    """
+    Returns the result of a regularised heavyside function of the
+    input value(s).
+    """
+    return 0.5 * (1. + 2./np.pi * np.arctan(x/eps))
+
+
+def _cv_delta(x, eps=1.):
+    """
+    Returns the result of a regularised dirac function of the
+    input value(s).
+    """
+    return eps / (eps**2 + x**2)
+
+
+def _cv_calculate_averages(img, Hphi):
+    """
+    Returns the average values 'inside' and 'outside'
```

Comment:
From my point, it would be better to run over the whole PR.
I personally prefer spaces everywhere, except (a) in function call signatures and (b) around `**`.